<ui version="4.0" >
 <author></author>
 <comment></comment>
 <exportmacro></exportmacro>
 <class>MessageLog</class>
 <widget class="QMainWindow" name="MessageLog" >
  <property name="geometry" >
   <rect>
    <x>0</x>
    <y>0</y>
    <width>835</width>
    <height>744</height>
   </rect>
  </property>
  <property name="font" >
   <font>
    <family>Arial</family>
    <pointsize>10</pointsize>
    <weight>50</weight>
    <italic>false</italic>
    <bold>false</bold>
    <underline>false</underline>
    <strikeout>false</strikeout>
   </font>
  </property>
  <property name="windowTitle" >
   <string>Recent Log Messages</string>
  </property>
  <property name="windowIcon" >
   <iconset resource="..\res\vidalia.qrc" >:/images/tor_logo16.png</iconset>
  </property>
  <property name="statusTip" >
   <string/>
  </property>
  <widget class="QWidget" name="centralwidget" >
   <layout class="QVBoxLayout" >
    <property name="margin" >
     <number>9</number>
    </property>
    <property name="spacing" >
     <number>6</number>
    </property>
    <item>
     <widget class="QTreeWidget" name="lstMessages" >
      <property name="contextMenuPolicy" >
       <enum>Qt::NoContextMenu</enum>
      </property>
      <property name="autoFillBackground" >
       <bool>false</bool>
      </property>
      <property name="alternatingRowColors" >
       <bool>true</bool>
      </property>
      <property name="selectionMode" >
       <enum>QAbstractItemView::ExtendedSelection</enum>
      </property>
      <property name="rootIsDecorated" >
       <bool>false</bool>
      </property>
      <property name="uniformRowHeights" >
       <bool>true</bool>
      </property>
      <property name="itemsExpandable" >
       <bool>false</bool>
      </property>
      <property name="sortingEnabled" >
       <bool>true</bool>
      </property>
      <column>
       <property name="text" >
        <string>Time</string>
       </property>
      </column>
      <column>
       <property name="text" >
        <string>Type</string>
       </property>
      </column>
      <column>
       <property name="text" >
        <string>Message</string>
       </property>
      </column>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" >
      <property name="margin" >
       <number>0</number>
      </property>
      <property name="spacing" >
       <number>6</number>
      </property>
      <item>
       <widget class="QPushButton" name="btnToggleSettings" >
        <property name="contextMenuPolicy" >
         <enum>Qt::NoContextMenu</enum>
        </property>
        <property name="toolTip" >
         <string>Toggles display of Message Log settings</string>
        </property>
        <property name="text" >
         <string>Show Settings</string>
        </property>
        <property name="checkable" >
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <spacer>
        <property name="orientation" >
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" >
         <size>
          <width>709</width>
          <height>29</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QFrame" name="frmSettings" >
      <property name="enabled" >
       <bool>true</bool>
      </property>
      <property name="sizePolicy" >
       <sizepolicy>
        <hsizetype>0</hsizetype>
        <vsizetype>0</vsizetype>
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize" >
       <size>
        <width>811</width>
        <height>181</height>
       </size>
      </property>
      <property name="contextMenuPolicy" >
       <enum>Qt::NoContextMenu</enum>
      </property>
      <property name="frameShape" >
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow" >
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QHBoxLayout" >
       <property name="margin" >
        <number>9</number>
       </property>
       <property name="spacing" >
        <number>6</number>
       </property>
       <item>
        <widget class="QGroupBox" name="groupBox" >
         <property name="contextMenuPolicy" >
          <enum>Qt::NoContextMenu</enum>
         </property>
         <property name="title" >
          <string>Tor Messages</string>
         </property>
         <layout class="QVBoxLayout" >
          <property name="margin" >
           <number>9</number>
          </property>
          <property name="spacing" >
           <number>6</number>
          </property>
          <item>
           <layout class="QVBoxLayout" >
            <property name="margin" >
             <number>0</number>
            </property>
            <property name="spacing" >
             <number>6</number>
            </property>
            <item>
             <widget class="QCheckBox" name="chkTorErr" >
              <property name="contextMenuPolicy" >
               <enum>Qt::NoContextMenu</enum>
              </property>
              <property name="toolTip" >
               <string/>
              </property>
              <property name="whatsThis" >
               <string/>
              </property>
              <property name="text" >
               <string>Tor Error</string>
              </property>
              <property name="icon" >
               <iconset resource="..\res\vidalia.qrc" >:/images/msg_error.png</iconset>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="chkTorWarn" >
              <property name="contextMenuPolicy" >
               <enum>Qt::NoContextMenu</enum>
              </property>
              <property name="toolTip" >
               <string/>
              </property>
              <property name="text" >
               <string>Tor Warning</string>
              </property>
              <property name="icon" >
               <iconset resource="..\res\vidalia.qrc" >:/images/msg_warn.png</iconset>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="chkTorNote" >
              <property name="contextMenuPolicy" >
               <enum>Qt::NoContextMenu</enum>
              </property>
              <property name="toolTip" >
               <string/>
              </property>
              <property name="text" >
               <string>Tor Notice</string>
              </property>
              <property name="icon" >
               <iconset resource="..\res\vidalia.qrc" >:/images/msg_notice.png</iconset>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="chkTorInfo" >
              <property name="contextMenuPolicy" >
               <enum>Qt::NoContextMenu</enum>
              </property>
              <property name="toolTip" >
               <string/>
              </property>
              <property name="text" >
               <string>Tor Info</string>
              </property>
              <property name="icon" >
               <iconset resource="..\res\vidalia.qrc" >:/images/msg_info.png</iconset>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="chkTorDebug" >
              <property name="contextMenuPolicy" >
               <enum>Qt::NoContextMenu</enum>
              </property>
              <property name="toolTip" >
               <string/>
              </property>
              <property name="text" >
               <string>Tor Debug</string>
              </property>
              <property name="icon" >
               <iconset resource="..\res\vidalia.qrc" >:/images/msg_debug.png</iconset>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <layout class="QVBoxLayout" >
         <property name="margin" >
          <number>0</number>
         </property>
         <property name="spacing" >
          <number>6</number>
         </property>
         <item>
          <widget class="QGroupBox" name="groupBox_2" >
           <property name="contextMenuPolicy" >
            <enum>Qt::NoContextMenu</enum>
           </property>
           <property name="title" >
            <string>Vidalia Messages</string>
           </property>
           <layout class="QVBoxLayout" >
            <property name="margin" >
             <number>9</number>
            </property>
            <property name="spacing" >
             <number>6</number>
            </property>
            <item>
             <layout class="QVBoxLayout" >
              <property name="margin" >
               <number>0</number>
              </property>
              <property name="spacing" >
               <number>6</number>
              </property>
              <item>
               <widget class="QCheckBox" name="chkVidErr" >
                <property name="contextMenuPolicy" >
                 <enum>Qt::NoContextMenu</enum>
                </property>
                <property name="toolTip" >
                 <string/>
                </property>
                <property name="text" >
                 <string>Vidalia Error</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QCheckBox" name="chkVidStat" >
                <property name="contextMenuPolicy" >
                 <enum>Qt::NoContextMenu</enum>
                </property>
                <property name="toolTip" >
                 <string/>
                </property>
                <property name="text" >
                 <string>Vidalia Status</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <spacer>
           <property name="orientation" >
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" >
            <size>
             <width>20</width>
             <height>71</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" >
         <property name="margin" >
          <number>0</number>
         </property>
         <property name="spacing" >
          <number>6</number>
         </property>
         <item>
          <widget class="QGroupBox" name="groupBox_3" >
           <property name="contextMenuPolicy" >
            <enum>Qt::NoContextMenu</enum>
           </property>
           <property name="title" >
            <string>Log History Size</string>
           </property>
           <layout class="QVBoxLayout" >
            <property name="margin" >
             <number>9</number>
            </property>
            <property name="spacing" >
             <number>6</number>
            </property>
            <item>
             <layout class="QHBoxLayout" >
              <property name="margin" >
               <number>0</number>
              </property>
              <property name="spacing" >
               <number>6</number>
              </property>
              <item>
               <widget class="QSpinBox" name="spnbxMaxCount" >
                <property name="minimumSize" >
                 <size>
                  <width>100</width>
                  <height>0</height>
                 </size>
                </property>
                <property name="contextMenuPolicy" >
                 <enum>Qt::NoContextMenu</enum>
                </property>
                <property name="toolTip" >
                 <string>Number of messages to keep in log</string>
                </property>
                <property name="maximum" >
                 <number>99999</number>
                </property>
                <property name="minimum" >
                 <number>1</number>
                </property>
               </widget>
              </item>
              <item>
               <spacer>
                <property name="orientation" >
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" >
                 <size>
                  <width>62</width>
                  <height>23</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <spacer>
           <property name="orientation" >
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" >
            <size>
             <width>20</width>
             <height>81</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" >
         <property name="margin" >
          <number>0</number>
         </property>
         <property name="spacing" >
          <number>6</number>
         </property>
         <item>
          <spacer>
           <property name="orientation" >
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" >
            <size>
             <width>101</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <layout class="QVBoxLayout" >
           <property name="margin" >
            <number>0</number>
           </property>
           <property name="spacing" >
            <number>6</number>
           </property>
           <item>
            <widget class="QPushButton" name="btnSaveSettings" >
             <property name="contextMenuPolicy" >
              <enum>Qt::NoContextMenu</enum>
             </property>
             <property name="toolTip" >
              <string>Saves the current Message Log settings</string>
             </property>
             <property name="text" >
              <string>Save Settings</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btnCancelSettings" >
             <property name="contextMenuPolicy" >
              <enum>Qt::NoContextMenu</enum>
             </property>
             <property name="toolTip" >
              <string>Cancels changes made to settings</string>
             </property>
             <property name="text" >
              <string>Cancel</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer>
             <property name="orientation" >
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" >
              <size>
               <width>20</width>
               <height>81</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar" >
   <property name="geometry" >
    <rect>
     <x>0</x>
     <y>0</y>
     <width>835</width>
     <height>19</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_Edit" >
    <property name="title" >
     <string>&amp;Edit</string>
    </property>
    <addaction name="actionCopy" />
    <addaction name="actionSelect_All" />
   </widget>
   <widget class="QMenu" name="menu_File" >
    <property name="title" >
     <string>&amp;File</string>
    </property>
    <addaction name="actionSave_All" />
    <addaction name="actionSave_Selected" />
    <addaction name="separator" />
    <addaction name="actionClose" />
   </widget>
   <widget class="QMenu" name="menu_Settings" >
    <property name="title" >
     <string>&amp;Settings</string>
    </property>
    <addaction name="actionMessage_Filters" />
    <addaction name="actionHistory_Size" />
   </widget>
   <widget class="QMenu" name="menu_View" >
    <property name="title" >
     <string>&amp;View</string>
    </property>
    <addaction name="actionClear" />
   </widget>
   <addaction name="menu_File" />
   <addaction name="menu_Edit" />
   <addaction name="menu_View" />
  </widget>
  <widget class="QStatusBar" name="statusbar" >
   <property name="geometry" >
    <rect>
     <x>0</x>
     <y>722</y>
     <width>835</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <action name="actionMessage_Filters" >
   <property name="text" >
    <string>Message Filters...</string>
   </property>
   <property name="statusTip" >
    <string>Set message filters</string>
   </property>
   <property name="font" >
    <font>
     <family>Arial</family>
     <pointsize>10</pointsize>
     <weight>50</weight>
     <italic>false</italic>
     <bold>false</bold>
     <underline>false</underline>
     <strikeout>false</strikeout>
    </font>
   </property>
   <property name="shortcut" >
    <string/>
   </property>
  </action>
  <action name="actionHistory_Size" >
   <property name="text" >
    <string>History Size...</string>
   </property>
   <property name="statusTip" >
    <string>Set maximum number of messages to display</string>
   </property>
   <property name="font" >
    <font>
     <family>Arial</family>
     <pointsize>10</pointsize>
     <weight>50</weight>
     <italic>false</italic>
     <bold>false</bold>
     <underline>false</underline>
     <strikeout>false</strikeout>
    </font>
   </property>
  </action>
  <action name="actionClear" >
   <property name="text" >
    <string>Clear History</string>
   </property>
   <property name="statusTip" >
    <string>Clear all messages from the Message Log</string>
   </property>
   <property name="font" >
    <font>
     <family>Arial</family>
     <pointsize>10</pointsize>
     <weight>50</weight>
     <italic>false</italic>
     <bold>false</bold>
     <underline>false</underline>
     <strikeout>false</strikeout>
    </font>
   </property>
   <property name="shortcut" >
    <string>Ctrl+E</string>
   </property>
  </action>
  <action name="actionCopy" >
   <property name="text" >
    <string>Copy</string>
   </property>
   <property name="statusTip" >
    <string>Copy the selected messages to the clipboard</string>
   </property>
   <property name="font" >
    <font>
     <family>Arial</family>
     <pointsize>10</pointsize>
     <weight>50</weight>
     <italic>false</italic>
     <bold>false</bold>
     <underline>false</underline>
     <strikeout>false</strikeout>
    </font>
   </property>
   <property name="shortcut" >
    <string>Ctrl+C</string>
   </property>
  </action>
  <action name="actionSelect_All" >
   <property name="text" >
    <string>Select All</string>
   </property>
   <property name="statusTip" >
    <string>Select all messages</string>
   </property>
   <property name="font" >
    <font>
     <family>Arial</family>
     <pointsize>10</pointsize>
     <weight>50</weight>
     <italic>false</italic>
     <bold>false</bold>
     <underline>false</underline>
     <strikeout>false</strikeout>
    </font>
   </property>
   <property name="shortcut" >
    <string>Ctrl+A</string>
   </property>
  </action>
  <action name="actionSave_All" >
   <property name="text" >
    <string>Save All</string>
   </property>
   <property name="statusTip" >
    <string>Save all messages to a file</string>
   </property>
   <property name="font" >
    <font>
     <family>Arial</family>
     <pointsize>10</pointsize>
     <weight>50</weight>
     <italic>false</italic>
     <bold>false</bold>
     <underline>false</underline>
     <strikeout>false</strikeout>
    </font>
   </property>
  </action>
  <action name="actionSave_Selected" >
   <property name="text" >
    <string>Save Selected</string>
   </property>
   <property name="statusTip" >
    <string>Save selected messages to a file</string>
   </property>
   <property name="font" >
    <font>
     <family>Arial</family>
     <pointsize>10</pointsize>
     <weight>50</weight>
     <italic>false</italic>
     <bold>false</bold>
     <underline>false</underline>
     <strikeout>false</strikeout>
    </font>
   </property>
  </action>
  <action name="actionClose" >
   <property name="text" >
    <string>Close</string>
   </property>
   <property name="statusTip" >
    <string>Close the Message Log</string>
   </property>
   <property name="font" >
    <font>
     <family>Arial</family>
     <pointsize>10</pointsize>
     <weight>50</weight>
     <italic>false</italic>
     <bold>false</bold>
     <underline>false</underline>
     <strikeout>false</strikeout>
    </font>
   </property>
   <property name="shortcut" >
    <string>Alt+F4</string>
   </property>
  </action>
 </widget>
 <pixmapfunction></pixmapfunction>
 <resources>
  <include location="res/vidalia.qrc" />
  <include location="../res/vidalia.qrc" />
 </resources>
 <connections>
  <connection>
   <sender>actionClose</sender>
   <signal>triggered()</signal>
   <receiver>MessageLog</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel" >
     <x>407</x>
     <y>378</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionClear</sender>
   <signal>triggered()</signal>
   <receiver>lstMessages</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel" >
     <x>407</x>
     <y>393</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSelect_All</sender>
   <signal>triggered()</signal>
   <receiver>lstMessages</receiver>
   <slot>selectAll()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel" >
     <x>407</x>
     <y>393</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnToggleSettings</sender>
   <signal>toggled(bool)</signal>
   <receiver>frmSettings</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>60</x>
     <y>728</y>
    </hint>
    <hint type="destinationlabel" >
     <x>418</x>
     <y>848</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
